source/Fleet.cpp

```diff
@@ -430,7 +430,10 @@ void Fleet::Enter(const System &system, list<shared_ptr<Ship>> &ships, const Pla
 		if(flagship)
 			ship->SetParent(flagship);
 		else
+		{
 			flagship = ship;
+			ship->SetFleetId(fleetId);
```

Comment:
If only the flagship has the fleetId, that means, if the flagship is gone, this fleet no longer contributes towards the cap, even if every other ship in the fleet is still present?
Is this the desired behaviour?

Reply:
Yes

Reply:
I could not find a better solution

Reply:
I suppose stepping through every ship in a system to check if there are any ships left with the fleetId of one being removed wouldn't be great for performance.

Reply:
Most uses I can think of for this feature would be to limit single ship fleets, or limit a particular flagship with normal escorts. I think that's an acceptable limitation.

Reply:
> I suppose stepping through every ship in a system to check if there are any ships left with the fleetId of one being removed wouldn't be great for performance.

Not really yes, i mean its bad for battles but when you are not in combat all ships of one fleet should leave simultan

Reply:
> Most uses I can think of for this feature would be to limit single ship fleets, or limit a particular flagship with normal escorts. I think that's an acceptable limitation.

Well if you do not care about accuracy then you can apply it even to larger fleets

Reply:
New idea:
You could check for the escorts of the dead ship make one of the escorts parent over the others and write the fleetId to it

Reply:
While this is faster then chhecking all ships it is still a huge process

Reply:
I agree with Terin and Warp that checking for flagship should be sufficient.

Doubly so if the alternative is a big performance hit.

Reply:
Another idea: we make a vector with pointers to all ships of the fleet. All we have to do then is checking if they are nullptr

Reply:
But i dont know if that would work